Refrigerator.cs

```diff
@@ -0,0 +1,238 @@
+            //}
+            freeSpace = Shelves.Sum(item => item.GetCurrentFreeSpace() + freeSpace);
+            return freeSpace;
+        }
+
+        //3
+        public bool AddItemForRefrigerator(Item item)
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                if (shelf.AddItemToShelf(item, shelf))
+                {
+                    Console.WriteLine("The product add succses to the fridge");
+                    return true;
+                }
+            }
+            Console.WriteLine("Their is not place in fridge");
+            return false;
+        }
+
+
+        //4
+        public Item RemoveItemForRefrigerator(int itemId)
+        {
+            Item item = new Item();
+            foreach (Shelf shelf in Shelves)
+            {
+                item = shelf.RemoveItemFromShelf(itemId);
+                if (item != null)
+                    break;
+            }
+            //item=(Item)Shelves.Where(s => s.REmoveItem(itemId)!=null);
+            return item;
+        }
+
+        //5*
+        public String CleanExpiredFromRefrigerator()
+        {
+            String ItemsCleanExpired = "";
+            foreach (Shelf shelf in Shelves)
+            {
+                ItemsCleanExpired += shelf.CleanExpiredFromShelf();
+            }
+            return ItemsCleanExpired;
+        }
+
+
+        //6
+        public List<Item> FindItemsByTypeAndKashrut(Kashrut kashrut, Type type)
+        {
+            List<Item> itemsTypeAndKashrut = new List<Item>();
+            foreach (Shelf shelf in Shelves)
+            {
+                itemsTypeAndKashrut.AddRange(shelf.FindItemsByTypeAndKashrut(kashrut, type));
+            }
+
+            return itemsTypeAndKashrut;
+        }
+
+
+        //7.1
+        public List<Item> SortItemsDescByExpirationDate()
+        {
+
+            List<Item> itemsSorted = new List<Item>();
+            foreach (Shelf shelf in Shelves)
+            {
+                itemsSorted.AddRange(shelf.Items);
+            }
+            itemsSorted.Sort((x, y) => x.ExpirationDate.CompareTo(y.ExpirationDate));
+            return itemsSorted;
+        }
+
+        //8.2
+        public List<Shelf> SortShelvesByFreeSpace(List<Shelf> shelves)
+        {
+            List<Shelf> shelvesSorted = new List<Shelf>();
+            shelvesSorted.AddRange(shelves);
+            shelvesSorted.Sort((x, y) => x.FreeSpace - y.FreeSpace);
+            return shelvesSorted;
+        }
+
+        public List<Refrigerator> SortRefrigeratorsByFreeSpace(List<Refrigerator> refrigerators)
+        {
+            List<Refrigerator> refrigeratorsSorted = refrigerators;
+            refrigeratorsSorted.Add(this);
+            refrigeratorsSorted.Sort((x, y) => y.GetFreeSpace() - x.GetFreeSpace());
+            return refrigeratorsSorted;
+        }
+
+        public void CleansByExpirationdateAndKosher(Kashrut kashrut, int countDay)
```

Comment:
Bug: This function removes items, even in cases where removing them will not give enough storage to go shopping